### ISSUE TITLE
Laravel 5.2

### ISSUE THREAD

**Issue by Bouhnosaure** [dingo/api#779]
Hi !

Thanks you 1000 times for your awesome work ! really great :smile:

Hum, Laravel 5.2 is released ! :thumbsup:

Have you plan the compatibility with this version ?

best regards 

Alex

**Comment by crynobone** [comment]
```json
"illuminate/routing": "~5.1",
"illuminate/support": "~5.1",
```

It's already supported, do you have problem installing it on 5.2?

**Comment by Stolz** [comment]
There are a few problems with L 5.2. For instance:

> Declaration of Dingo\Api\Http\Response::cookie(Symfony\Component\HttpFoundation\Cookie $cookie) should be compatible with Illuminate\Http\Response::cookie($cookie)

**Comment by arthurkirkosa** [comment]
Problem 1
    - Installation request for dingo/api 1.0.x-dev -> satisfiable by dingo/api[1.0.x-dev].
    - Conclusion: remove laravel/framework v5.2.0
    - dingo/api 1.0.x-dev requires dingo/blueprint 0.1.* -> satisfiable by dingo/blueprint[v0.1.0, v0.1.1, v0.1.2, v0.1.3].
    - dingo/blueprint v0.1.0 requires illuminate/support 5.1.* -> satisfiable by illuminate/support[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.6, v5.1.8].
    - dingo/blueprint v0.1.1 requires illuminate/support 5.1.* -> satisfiable by illuminate/support[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.6, v5.1.8].
    - dingo/blueprint v0.1.2 requires illuminate/support 5.1.* -> satisfiable by illuminate/support[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.6, v5.1.8].
    - dingo/blueprint v0.1.3 requires illuminate/support 5.1.* -> satisfiable by illuminate/support[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.6, v5.1.8].
    - don't install illuminate/support v5.1.1|don't install laravel/framework v5.2.0
    - don't install illuminate/support v5.1.13|don't install laravel/framework v5.2.0
    - don't install illuminate/support v5.1.16|don't install laravel/framework v5.2.0
    - don't install illuminate/support v5.1.2|don't install laravel/framework v5.2.0
    - don't install illuminate/support v5.1.20|don't install laravel/framework v5.2.0
    - don't install illuminate/support v5.1.22|don't install laravel/framework v5.2.0
    - don't install illuminate/support v5.1.25|don't install laravel/framework v5.2.0
    - don't install illuminate/support v5.1.6|don't install laravel/framework v5.2.0
    - don't install illuminate/support v5.1.8|don't install laravel/framework v5.2.0
    - Installation request for laravel/framework 5.2.* -> satisfiable by laravel/framework[v5.2.0].

**Comment by crynobone** [comment]
It's failing on `dingo/blueprint` and not `dingo/api`.

```
- dingo/blueprint v0.1.0 requires illuminate/support 5.1.* -> satisfiable by illuminate/support[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.6, v5.1.8].
    - dingo/blueprint v0.1.1 requires illuminate/support 5.1.* -> satisfiable by illuminate/support[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.6, v5.1.8].
    - dingo/blueprint v0.1.2 requires illuminate/support 5.1.* -> satisfiable by illuminate/support[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.6, v5.1.8].
    - dingo/blueprint v0.1.3 requires illuminate/support 5.1.* -> satisfiable by illuminate/support[v5.1.1, v5.1.13, v5.1.16, v5.1.2, v5.1.20, v5.1.22, v5.1.25, v5.1.6, v5.1.8].
```

<https://github.com/dingo/blueprint/compare/v0.1.3...master>

**Comment by arthurkirkosa** [comment]
True ... https://github.com/dingo/blueprint/issues/22

**Comment by stueynet** [comment]
+1 for this. I opened the issue in Blue print a few days ago

https://github.com/dingo/blueprint/issues/22

**Comment by irazasyed** [comment]
+1 have the compatibility issue.

**Comment by irazasyed** [comment]
Opened a PR #780 which fixes the compatibility issue.

**Comment by LukasMa** [comment]
:+1: thanks @irazasyed Hope this get's merged soon, wan't to update as fast as possible :)

**Comment by jasonlewis** [comment]
This should now be resolved but I've not done any testing on Laravel 5.2 so use at your own risk.

I'll be back on track with Dingo early in the new year. Sorry for all the delays and setbacks guys and gals! And thanks for the continued usage and support, means a lot!

**Comment by LukasMa** [comment]
Just letting you know:

"message": "Token Signature could not be verified.",
"status_code": 401,
"debug": {
    "line": 50,
    "file": "vendor\dingo\api\src\Auth\Provider\JWT.php",
    "class": "Symfony\Component\HttpKernel\Exception\UnauthorizedHttpException",

At least for me, authentication with JWTs does not work after upgrading to 5.2.

**Comment by irazasyed** [comment]
@LukasMa JWT Package still doesn't support Laravel 5.2. @crynobone had sent a PR to add support but it is yet to be merged.

**Comment by LukasMa** [comment]
@irazasyed Thanks for letting me know. Wasn't quiet sure if the error is caused by JWT or Dingo.

**Comment by irazasyed** [comment]
@LukasMa No probs. Once it gets merged, you can then test again and open another issue ticket if the issue is with Dingo :)

**Comment by LukasMa** [comment]
@irazasyed Can you link the PR you mentioned? Couldn't find it. Furhtermore, I found the issue #783 which is now cloesd. The author solved the issue with upgrading to php >5.6.4. But current stable versions are 5.6.16 and 7.0.1. I tried it with both and did not have success.

**Comment by irazasyed** [comment]
@LukasMa https://github.com/tymondesigns/jwt-auth/pull/347

**Comment by LukasMa** [comment]
Ah yes, that was it. Thanks

But I checkd again: I'm using up to date versions of dingo and jwt, tried with php 5.6.16 and php 7.0.0. But still: I can't get it to work. Requesting the token works, but I can't authenticate with it. I always get the "401: Token Signature could not be verified." error. I tried to use jwt-auth middleware to verify my token, no luck. JWT won't validate the token I created 1 minute before.
